### PR TITLE
[NOX] Implementing loading states to the new settings pages

### LINKED ISSUE #52955: [NOX]: Implement Loading/Empty states
### Description

Implement loading and empty states for the Payments Settings page in line with the designs

### Acceptance criteria

* Loading states should show in line with the designs when the payment gateways are being loaded
* When a gateway is added/removed from the list, or a re-render happens, the loading state should appear

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Dev notes

[This selector](https://github.com/woocommerce/woocommerce/blob/3bf55034002a61ba5d5462e7b53f98541cbccb21/packages/js/data/src/payment-settings/selectors.ts#L37-L37) can be used to detect loading:

```js
export function isFetching( state: PaymentSettingsState ): boolean {
	return state.isFetching || false;
}
```

### PR DESCRIPTION
**Pull request by dmallory42**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #52955 

### Screenshots

Screenshot 2024-11-23 at 07 21 05

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure you have enabled the new Reactified settings pages (see the instructions in https://github.com/woocommerce/woocommerce/pull/52824 if you haven't done this previously).
1. Run ` pnpm run --filter=@woocommerce/plugin-woocommerce watch:build`
2. Load the settings page, you should see the loadable state as shown in the screenshot appear before the content loads.
3. Install/Enable a recommended payment gateway (e.g. one of the top 2 choices that appear in the payments gateways list, or one of the options that appear in `Other payment options`. You should see the loadable state briefly appear while we add/remove options from the main payment gateways list.
4. Replace [this block of code](https://github.com/woocommerce/woocommerce/blob/3b891b7f0ddc0557057744c294557727cb6d8060/plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx#L137) as follows:

 ```ts
			{ isFetching ? (
				<ListPlaceholder rows={ 5 } />
			) : (
				<ListPlaceholder rows={ 5 } />
			) }
```

5. Reload the page, the loading state should appear at all times. Verify it matches the designs.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Add loading states to the React settings page
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.scss [hunk 1]
```diff
@@ -114,6 +114,13 @@ $font-sf-pro-display: "SF Pro Text", -apple-system, BlinkMacSystemFont, "Segoe U
 		}
 	}
 
+	.other-payment-gateways__header__title__image-placeholder {
```
Comment (oaratovskyi) [inline plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.scss:117]: @dmallory42 what is this about? I guess we can remove it, I don't see where it's used

**Review by oaratovskyi — APPROVED** [review]
LGTM and works well! Nothing blocking, left one comment :shipit:

File: plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.scss (hunk 1 above)
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.scss:117]: I see where it's used!

**Comment by oaratovskyi** [comment]
@dmallory42 I've added fetching placeholder for other payment gateways in expanded state, see https://github.com/woocommerce/woocommerce/pull/53122/commits/3056b1c002881da819dba8e8d5adc877dd67d642 and https://github.com/woocommerce/woocommerce/pull/53122/commits/b289f041c1a81348b6af13a6af22e82b1ede9583
image

**Comment by oaratovskyi** [comment]
Offline look like this 
image

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

**Pull request merged**